src/server.cpp

```diff
@@ -1671,17 +1671,68 @@ void Server::SendHUDSetParam(session_t peer_id, u16 param, const std::string &va
 	Send(&pkt);
 }
 
-void Server::SendSetSky(session_t peer_id, const video::SColor &bgcolor,
-		const std::string &type, const std::vector<std::string> &params,
-		bool &clouds)
+void Server::SendSetSky(session_t peer_id, const SkyboxParams &params)
 {
 	NetworkPacket pkt(TOCLIENT_SET_SKY, 0, peer_id);
-	pkt << bgcolor << type << (u16) params.size();
 
-	for (const std::string &param : params)
-		pkt << param;
+	// Handle prior clients here
+	if (m_clients.getProtocolVersion(peer_id) < 38) {
+		pkt << params.bgcolor << params.type << (u16) params.params.size();
+
+		for(size_t i=0; i<params.params.size(); i++)
+			pkt << params.params[i];
+
+		pkt << params.clouds;
+
+		Send(&pkt);	
```

Comment:
`Send()` is called outside the if, this one is duplicate

Reply:
Ah. I missed that one.

Reply:
Seems unresolved?

Reply:
Mistake was resolved when I added backwards compatibility.

Reply:
Now fixed.